### ISSUE TITLE
[ICE] Out of bounds check crash when checking impl blocks with `#[cfg(..)]` in generics

### ISSUE THREAD

**Issue by 5GameMaker** [rust-lang/rust#154562]
I tried this code:

```rust
struct Test<#[cfg(feature = "alloc")] A> {
    #[cfg(feature = "alloc")]
    alloc: A,
}
impl<#[cfg(feature = "alloc")] A> Test<A> {}
```

I expected to see this happen: An error saying that that `A` is not defined.

Instead, this happened: ICE crash

### Meta

`rustc --version --verbose`:
```
rustc 1.96.0-nightly (562dee482 2026-03-21)
binary: rustc
commit-hash: 562dee4820c458d823175268e41601d4c060588a
commit-date: 2026-03-21
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.0
```

<details><summary>Backtrace</summary>
<p>

```
thread 'rustc' (36825) panicked at /rustc-dev/562dee4820c458d823175268e41601d4c060588a/compiler/rustc_resolve/src/late/diagnostics.rs:3384:36:
index out of bounds: the len is 0 but the index is 0
stack backtrace:
   0:     0x7fdce478455b - <<std[31e582c13f4e914f]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[fa0eadad68403074]::fmt::Display>::fmt
   1:     0x7fdce4e2e8c8 - core[fa0eadad68403074]::fmt::write
   2:     0x7fdce479b566 - <std[31e582c13f4e914f]::sys::stdio::unix::Stderr as std[31e582c13f4e914f]::io::Write>::write_fmt
   3:     0x7fdce475a568 - std[31e582c13f4e914f]::panicking::default_hook::{closure#0}
   4:     0x7fdce4777963 - std[31e582c13f4e914f]::panicking::default_hook
   5:     0x7fdce3773dbc - std[31e582c13f4e914f]::panicking::update_hook::<alloc[49b8922c36ca62de]::boxed::Box<rustc_driver_impl[5d9f716931f6bd73]::install_ice_hook::{closure#1}>>::{closure#0}
   6:     0x7fdce4777c42 - std[31e582c13f4e914f]::panicking::panic_with_hook
   7:     0x7fdce475a628 - std[31e582c13f4e914f]::panicking::panic_handler::{closure#0}
   8:     0x7fdce474eb79 - std[31e582c13f4e914f]::sys::backtrace::__rust_end_short_backtrace::<std[31e582c13f4e914f]::panicking::panic_handler::{closure#0}, !>
   9:     0x7fdce475c08d - __rustc[fc8e3a9f23f26cb]::rust_begin_unwind
  10:     0x7fdce17aaf7c - core[fa0eadad68403074]::panicking::panic_fmt
  11:     0x7fdce16ccdb2 - core[fa0eadad68403074]::panicking::panic_bounds_check
  12:     0x7fdce42f8dff - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>::detect_and_suggest_const_parameter_error
  13:     0x7fdce1202c87 - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>::smart_resolve_path_fragment::{closure#0}
  14:     0x7fdce5aaa13b - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>::smart_resolve_path_fragment
  15:     0x7fdce5aa80e9 - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor as rustc_ast[270aa3e6685cba48]::visit::Visitor>::visit_ty
  16:     0x7fdce5aa8432 - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor as rustc_ast[270aa3e6685cba48]::visit::Visitor>::visit_ty
  17:     0x7fdce5ac02b4 - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>::resolve_item
  18:     0x7fdce5abefc3 - rustc_ast[270aa3e6685cba48]::visit::visit_items::<rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>
  19:     0x7fdce5ac4910 - <rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>::resolve_item
  20:     0x7fdce5abefc3 - rustc_ast[270aa3e6685cba48]::visit::visit_items::<rustc_resolve[27e364e54eab9eed]::late::LateResolutionVisitor>
  21:     0x7fdce53a5e64 - <rustc_resolve[27e364e54eab9eed]::Resolver>::resolve_crate::{closure#0}
  22:     0x7fdce53a3947 - <rustc_resolve[27e364e54eab9eed]::Resolver>::resolve_crate
  23:     0x7fdce538aa89 - rustc_interface[fa77fbe53284b165]::passes::configure_and_expand
  24:     0x7fdce60a7378 - rustc_interface[fa77fbe53284b165]::passes::resolver_for_lowering_raw
  25:     0x7fdce60a70f3 - rustc_query_impl[787df1eb52488c09]::query_impl::resolver_for_lowering_raw::invoke_provider_fn::__rust_begin_short_backtrace
  26:     0x7fdce62ced42 - rustc_query_impl[787df1eb52488c09]::execution::try_execute_query::<rustc_middle[bc4b12ab5fd100e2]::query::caches::SingleCache<rustc_middle[bc4b12ab5fd100e2]::query::erase::ErasedData<[u8; 16usize]>>, true>
  27:     0x7fdce62ce67b - rustc_query_impl[787df1eb52488c09]::query_impl::resolver_for_lowering_raw::execute_query_incr::__rust_end_short_backtrace
  28:     0x7fdce61320be - rustc_interface[fa77fbe53284b165]::interface::run_compiler::<(), rustc_driver_impl[5d9f716931f6bd73]::run_compiler::{closure#0}>::{closure#1}
  29:     0x7fdce6043f3e - std[31e582c13f4e914f]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[fa77fbe53284b165]::util::run_in_thread_with_globals<rustc_interface[fa77fbe53284b165]::util::run_in_thread_pool_with_globals<rustc_interface[fa77fbe53284b165]::interface::run_compiler<(), rustc_driver_impl[5d9f716931f6bd73]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  30:     0x7fdce60447e0 - <std[31e582c13f4e914f]::thread::lifecycle::spawn_unchecked<rustc_interface[fa77fbe53284b165]::util::run_in_thread_with_globals<rustc_interface[fa77fbe53284b165]::util::run_in_thread_pool_with_globals<rustc_interface[fa77fbe53284b165]::interface::run_compiler<(), rustc_driver_impl[5d9f716931f6bd73]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[fa0eadad68403074]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  31:     0x7fdce604566c - <std[31e582c13f4e914f]::sys::thread::unix::Thread>::new::thread_start
  32:     0x7fdcdfaa597a - <unknown>
  33:     0x7fdcdfb292bc - <unknown>
  34:                0x0 - <unknown>

error: the compiler unexpectedly panicked. This is a bug

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: please attach the file at `/home/buj/Projects/rust/rtmp-server/rustc-ice-2026-03-29T20_45_52-36823.txt` to your bug report

note: rustc 1.96.0-nightly (562dee482 2026-03-21) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2 -C incremental=[REDACTED]

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [resolver_for_lowering_raw] getting the resolver for lowering
end of query stack
```

</p>
</details>

**Comment by chenyukang** [comment]
it's a duplicate of https://github.com/rust-lang/rust/issues/154165, already fixed.
